source/material_model/reactive_fluid_transport.cc

```diff
@@ -0,0 +1,376 @@
+    void
+    ReactiveFluidTransport<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Reactive Fluid Transport Model");
+        {
+          AssertThrow( prm.get("Base model") != "reactive fluid transport",
+                       ExcMessage("You may not use ``reactive fluid transport'' "
+                                  "as the base model for the reactive fluid transport "
+                                  "model itself.") );
+
+          reference_rho_f                   = prm.get_double ("Reference fluid density");
+          shear_to_bulk_viscosity_ratio     = prm.get_double ("Shear to bulk viscosity ratio");
+          eta_f                             = prm.get_double ("Reference fluid viscosity");
+          reference_permeability            = prm.get_double ("Reference permeability");
+          alpha_phi                         = prm.get_double ("Exponential fluid weakening factor");
+          fluid_compressibility             = prm.get_double ("Fluid compressibility");
+          fluid_reaction_time_scale         = prm.get_double ("Fluid reaction time scale for operator splitting");
+
+          // Create the base model and initialize its SimulatorAccess base
+          // class; it will get a chance to read its parameters below after we
+          // leave the current section.
+          base_model = create_material_model<dim>(prm.get("Base model"));
+          if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(base_model.get()))
+            sim->initialize_simulator (this->get_simulator());
+
+          if (this->convert_output_to_years() == true)
+            fluid_reaction_time_scale *= year_in_seconds;
+
+          // Reaction scheme parameter
+          if (prm.get ("Fluid-solid reaction scheme") == "zero solubility")
+            fluid_solid_reaction_scheme = zero_solubility;
+          else if (prm.get ("Fluid-solid reaction scheme") == "no reaction")
+            fluid_solid_reaction_scheme = no_reaction;
+          else
+            AssertThrow(false, ExcMessage("Not a valid fluid-solid reaction scheme"));
+
+          if (prm.get ("Fluid-solid reaction scheme") == "no reaction")
+            {
+              AssertThrow(this->get_parameters().use_operator_splitting == false,
+                          ExcMessage("The Fluid-reaction scheme no reaction should not be used with operator splitting."));
+            }
+
+          if (prm.get ("Fluid-solid reaction scheme") == "zero solubility")
+            {
+              AssertThrow(this->get_parameters().use_operator_splitting,
+                          ExcMessage("The Fluid-reaction scheme zero solubility must be used with operator splitting."));
+              AssertThrow(this->get_material_model().is_compressible() == false,
+                          ExcMessage("The Fluid-reaction scheme zero solubility must be used with an incompressible base model."));
```

Comment:
This does not work well for all models, since at this point we have not parsed the parameters from the base model. So if the base model has a compressibility that depends on an input parameter, at this point in the code we will not know about it. It would be better to move this after parsing the parameters from the base model or ito the initialize() function.